tests/testsuite/tool_paths.rs

```diff
@@ -352,3 +352,19 @@ fn cfg_ignored_fields() {
         )
         .run();
 }
+
+#[cargo_test]
+fn env_var_contain_lower_case() {
+    let p = project().file("src/main.rs", "fn main() {}").build();
+
+    let env_key = "CARGO_TARGET_X86_64_UNKNOWN_LINUX_musl_LINKER";
```

Comment:
For the sake of matching the likely use case, could you run two variants of this test, one with the key you have, and one with the key `CARGO_TARGET_x86_64_unknown_linux_musl_LINKER` ? (The latter seems more likely to occur.)